tests/unit/lxd/test_launcher.py

```diff
@@ -41,22 +42,35 @@ def mock_lxc():
 
 
 @pytest.fixture
-def mock_lxd_instance():
-    with mock.patch(
+def fake_instance():
+    """Returns a fake LXD Instance"""
+    instance = MagicMock()
+    # the name has an invalid character to ensure the instance_name will be different
+    # so the two are not conflated in the unit tests
+    instance.name = "test-instance-$"
+    instance.instance_name = "test-instance-fa2d407652a1c51f6019"
+    instance.project = "test-project"
+    instance.remote = "test-remote"
+    instance.exists.return_value = False
+    instance.is_running.return_value = False
+    return instance
+
+
+@pytest.fixture
+def mock_lxd_instance(fake_instance, mocker):
+    """Mock LXDInstance and return a fake instance."""
+    yield mocker.patch(
         "craft_providers.lxd.launcher.LXDInstance",
         spec=lxd.LXDInstance,
-    ) as mock_instance:
-        mock_instance.return_value.name = "test-instance-$"
-        # the name has an invalid character, so the instance_name will be different
-        mock_instance.return_value.instance_name = "test-instance-fa2d407652a1c51f6019"
-        mock_instance.return_value.project = "test-project"
-        mock_instance.return_value.remote = "test-remote"
-        yield mock_instance
-
+        # single element list appears accidental but side_effect must be an interable.
```

Comment:
Is this a typo for iterable?

Reply:
Yes, thanks.  Interesting that codespell doesn't catch it.